mlrun/runtimes/nuclio/application/application.py

```diff
@@ -344,24 +331,18 @@ def deploy(
             builder_env=builder_env,
         )
         logger.info(
-            "Successfully deployed function, creating API gateway",
-            api_gateway_name=self.status.api_gateway_name,
-            authentication_mode=authentication_mode,
+            "Successfully deployed function.",
         )
 
         # Restore the source in case it was removed to make nuclio not consider it when building
         if not self.spec.build.source and self.status.application_source:
             self.spec.build.source = self.status.application_source
         self.save(versioned=False)
 
-        ports = self.spec.internal_application_port if direct_port_access else []
-        return self.create_api_gateway(
-            name=self.status.api_gateway_name,
-            ports=ports,
-            authentication_mode=authentication_mode,
-            authentication_creds=authentication_creds,
-            ssl_redirect=ssl_redirect,
+        logger.info(
+            "Application online, proceed to API gateway creation to make it accessible."
```

Comment:
this might need to be a warning to "pop out" to the user.
Something like
```suggestion
        logger.warn(
            "Application is online but is not accessible. Use the `create_api_gateway` method to make it accessible."
```